
<!DOCTYPE HTML>

<html lang="en" class="_theme-default">

<head>
  <meta charset="utf-8">
  <title>Std&#58;&#58;underflow_error - C++ - W3cubDocs</title>
  
  <meta name="description" content="Defines a type of object to be thrown as exception. It may be used to report arithmetic underflow errors (that is, situations where the result of a &hellip;">
  <meta name="keywords" content="std, underflow, error, c++, cpp">
  <meta name="HandheldFriendly" content="True">
  <meta name="MobileOptimized" content="320">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="mobile-web-app-capable" content="yes">
  
  <link rel="canonical" href="https://docs.w3cub.com/cpp/error/underflow_error/underflow_error.html">
  <link href="/favicon.png" rel="icon">
  <link rel="stylesheet" type="text/css" href="/assets/application-e498cd0ebe8746846fec95b1a53ab3bb0fb7f47f794f0a38f44c98a1f0d03b21d777ae2c583732e44a5a890f6eacb79a5333545db9d5f3616091ba21ca17d916.css">
  <script src="/assets/application-79c555f6b25481fffac2cac30a7f3e54e608ca09e9e8e42bb1790095ba6d0fcace47d6bc624ddce952c70370892f2d46864f89e6943d4f7f7ff16c8a3231a91a.js" type="text/javascript"></script>
  <script src="/json/cpp.js"></script>
  
<!-- Google tag (gtag.js) -->
<script async src="https://www.googletagmanager.com/gtag/js?id=G-R3WC07G3GB"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());
  gtag('config', 'G-R3WC07G3GB');
</script>
<script async src="https://pagead2.googlesyndication.com/pagead/js/adsbygoogle.js?client=ca-pub-2572770204602497"
     crossorigin="anonymous"></script>
<script async custom-element="amp-auto-ads"
  src="https://cdn.ampproject.org/v0/amp-auto-ads-0.1.js">
</script>


</head>

<body class="docs">
	<amp-auto-ads type="adsense"
              data-ad-client="ca-pub-2572770204602497">
	</amp-auto-ads>
	<div class="_app">
	<header class="_header">

  <a class="_home-link" href="/" ></a>
  <a class="_menu-link"></a>
  <h1 class="_logo">
    <a href="/" class="_nav-link" title="API Documentation Browser">W3cubDocs</a>
  </h1>
  
  <span class="_logo-sub-nav">/</span><span class="_logo-sub-nav"><a href="/cpp/" class="_nav-link" title="" style="margin-left:0;">C++</a></span>
  
  <nav class="_nav">
    <a href="https://tools.w3cub.com/?_sp=docs" target="_blank" class="_nav-link ">W3cubTools</a>
    <a href="/cheatsheets/" class="_nav-link ">Cheatsheets</a>
    <a href="/about/" class="_nav-link ">About</a>
  </nav>
</header>
	<section class="_sidebar">
		
		<form class="_search">
		  <input type="search" class="_search-input" placeholder="Search&hellip;" autocomplete="off" autocapitalize="off" autocorrect="off" spellcheck="false" maxlength="20">
		  <a class="_search-clear"></a>
		  <div class="_search-tag"></div>
		</form>
		
		<div class="_list-wrap">
			<div class="_list">
			
			</div>
		</div>
	</section>
	<section class="_container ">
		<div class="_content">
			<ins class="adsbygoogle"
     style="display:block"
     data-ad-client="ca-pub-2572770204602497"
     data-ad-slot="6861657091"
     data-ad-format="auto"
     data-full-width-responsive="true"></ins>
<script>
     (adsbygoogle = window.adsbygoogle || []).push({});
</script>
			<div class="_page _c">
				
				
<h1 id="firstHeading" class="firstHeading">std::underflow_error</h1>            <table class="t-dcl-begin"> <tr class="t-dsc-header"> <th> Defined in header <code><a href="../../header/stdexcept" title="cpp/header/stdexcept">&lt;stdexcept&gt;</a></code> </th> <th> </th> <th> </th> </tr> <tr class="t-dcl"> <td class="t-dcl-nopad"> <pre data-language="cpp">class underflow_error;</pre>
</td> <td class="t-dcl-nopad"> </td> <td class="t-dcl-nopad"> </td> </tr>  </table> <p>Defines a type of object to be thrown as exception. It may be used to report arithmetic underflow errors (that is, situations where the result of a computation is a subnormal floating-point value).</p>
<p>The standard library components do not throw this exception (mathematical functions report underflow errors as specified in <code><a href="../../numeric/math/math_errhandling" title="cpp/numeric/math/math errhandling">math_errhandling</a></code>). Third-party libraries, however, use this. For example, <a rel="nofollow" class="external text" href="http://www.boost.org/doc/libs/1_66_0/libs/math/doc/html/math_toolkit/error_handling.html">boost.math</a> throws <code>std::underflow_error</code> if <code>boost::math::policies::throw_on_error</code> is enabled (the default setting).</p>
<div class="t-inheritance-diagram"> <div class="center"><div class="noresize" style="height: 40px; width: 385px; ">
<map name="ImageMap_1_166655361"><area href="../exception" shape="rect" coords="5,4,107,36" alt="cpp/error/exception" title="cpp/error/exception">
<area href="../runtime_error/runtime_error" shape="rect" coords="135,4,238,36" alt="cpp/error/runtime error" title="cpp/error/runtime error"></map><img alt="std-underflow error-inheritance.svg" src="data:image/svg+xml;base64,PD94bWwgdmVyc2lvbj0iMS4wIiBlbmNvZGluZz0iVVRGLTgiIHN0YW5kYWxvbmU9Im5vIj8+CjwhRE9DVFlQRSBzdmcgUFVCTElDICItLy9XM0MvL0RURCBTVkcgMS4xLy9FTiIgImh0dHA6Ly93d3cudzMub3JnL0dyYXBoaWNzL1NWRy8xLjEvRFREL3N2ZzExLmR0ZCI+CjwhLS0gR2VuZXJhdGVkIGJ5IGdyYXBodml6IHZlcnNpb24gMi4yNi4zICgyMDEwMDEyNi4xNjAwKQogLS0+CjwhLS0gVGl0bGU6IGluaGVyaXRhbmNlIFBhZ2VzOiAxIC0tPgo8c3ZnIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgeG1sbnM6eGxpbms9Imh0dHA6Ly93d3cudzMub3JnLzE5OTkveGxpbmsiIHdpZHRoPSIzODVweCIgaGVpZ2h0PSI0MHB4IiB2aWV3Qm94PSIwLjAwIDAuMDAgNDI2LjAwIDQ0LjAwIj4KPGcgaWQ9ImdyYXBoMSIgY2xhc3M9ImdyYXBoIiB0cmFuc2Zvcm09InJvdGF0ZSgwKSB0cmFuc2xhdGUoNCA0MCkiPgo8dGl0bGU+aW5oZXJpdGFuY2U8L3RpdGxlPgo8IS0tIGUxIC0tPgo8ZyBpZD0ibm9kZTEiIGNsYXNzPSJub2RlIj48dGl0bGU+ZTE8L3RpdGxlPgo8YT4KPHBvbHlnb24gZmlsbD0id2hpdGUiIHN0cm9rZT0iYmxhY2siIHBvaW50cz0iMTE1LC0zNiAxLC0zNiAxLC0xLjc3NjM2ZS0xNCAxMTUsLTMuNTUyNzFlLTE1IDExNSwtMzYiLz4KPHRleHQgdGV4dC1hbmNob3I9Im1pZGRsZSIgeD0iNTgiIHk9Ii0xNC40IiBmb250LWZhbWlseT0iRGVqYVZ1IFNhbnMsIHZlcmRhbmEsIHNhbnMtc2VyaWYiIGZvbnQtc2l6ZT0iMTQuMDAiPmV4Y2VwdGlvbjwvdGV4dD4KPC9hPgo8L2c+CjwhLS0gZTIgLS0+CjxnIGlkPSJub2RlMiIgY2xhc3M9Im5vZGUiPjx0aXRsZT5lMjwvdGl0bGU+CjxhPgo8cG9seWdvbiBmaWxsPSJ3aGl0ZSIgc3Ryb2tlPSJibGFjayIgcG9pbnRzPSIyNjAsLTM2IDE0NiwtMzYgMTQ2LC0xLjc3NjM2ZS0xNCAyNjAsLTMuNTUyNzFlLTE1IDI2MCwtMzYiLz4KPHRleHQgdGV4dC1hbmNob3I9Im1pZGRsZSIgeD0iMjAzIiB5PSItMTQuNCIgZm9udC1mYW1pbHk9IkRlamFWdSBTYW5zLCB2ZXJkYW5hLCBzYW5zLXNlcmlmIiBmb250LXNpemU9IjE0LjAwIj5ydW50aW1lX2Vycm9yPC90ZXh0Pgo8L2E+CjwvZz4KPCEtLSBlMiYjNDU7Jmd0O2UxIC0tPgo8ZyBpZD0iZWRnZTIiIGNsYXNzPSJlZGdlIj48dGl0bGU+ZTItJmd0O2UxPC90aXRsZT4KPHBhdGggZmlsbD0ibm9uZSIgc3Ryb2tlPSJibGFjayIgZD0iTTE0NS4yNzYsLTE4QzE0MC41MzIsLTE4IDEzNS43LC0xOCAxMzAuODY3LC0xOCIvPgo8cG9seWdvbiBmaWxsPSJub25lIiBzdHJva2U9ImJsYWNrIiBwb2ludHM9IjEzMC41MjYsLTEyLjc1MDEgMTE1LjUyNiwtMTggMTMwLjUyNiwtMjMuMjUwMSAxMzAuNTI2LC0xMi43NTAxIi8+CjwvZz4KPCEtLSBlMyAtLT4KPGcgaWQ9Im5vZGUzIiBjbGFzcz0ibm9kZSI+PHRpdGxlPmUzPC90aXRsZT4KPGE+Cjxwb2x5Z29uIGZpbGw9ImxpZ2h0Z3JleSIgc3Ryb2tlPSJibGFjayIgcG9pbnRzPSI0MTgsLTM2IDI5MCwtMzYgMjkwLC0xLjc3NjM2ZS0xNCA0MTgsLTMuNTUyNzFlLTE1IDQxOCwtMzYiLz4KPHRleHQgdGV4dC1hbmNob3I9Im1pZGRsZSIgeD0iMzU0IiB5PSItMTQuNCIgZm9udC1mYW1pbHk9IkRlamFWdSBTYW5zLCB2ZXJkYW5hLCBzYW5zLXNlcmlmIiBmb250LXNpemU9IjE0LjAwIj51bmRlcmZsb3dfZXJyb3I8L3RleHQ+CjwvYT4KPC9nPgo8IS0tIGUzJiM0NTsmZ3Q7ZTIgLS0+CjxnIGlkPSJlZGdlNCIgY2xhc3M9ImVkZ2UiPjx0aXRsZT5lMy0mZ3Q7ZTI8L3RpdGxlPgo8cGF0aCBmaWxsPSJub25lIiBzdHJva2U9ImJsYWNrIiBkPSJNMjg5Ljk2MiwtMThDMjg1LjQ0MSwtMTggMjgwLjg2MywtMTggMjc2LjI5NywtMTgiLz4KPHBvbHlnb24gZmlsbD0ibm9uZSIgc3Ryb2tlPSJibGFjayIgcG9pbnRzPSIyNzUuOTA4LC0xMi43NTAxIDI2MC45MDgsLTE4IDI3NS45MDgsLTIzLjI1MDEgMjc1LjkwOCwtMTIuNzUwMSIvPgo8L2c+CjwvZz4KPC9zdmc+Cg==" width="385" height="40" usemap="#ImageMap_1_166655361">
</div></div> <div style="text-align:center;"> <p>Inheritance diagram.</p>
</div> </div> <h3 id="Member_functions">Member functions</h3> <table class="t-dsc-begin"> <tr class="t-dsc"> <td> <div><strong class="selflink"> <span class="t-lines"><span>(constructor)</span></span></strong></div> </td> <td> constructs the exception object <br> <span class="t-mark">(public member function)</span> </td>
</tr> </table> 
<div class="t-member"> <h2 id="std::underflow_error::underflow_error">  std::underflow_error::underflow_error </h2> <table class="t-dcl-begin"> <tr class="t-dcl"> <td> <pre data-language="cpp">explicit underflow_error( const std::string&amp; what_arg );</pre>
</td> <td> (1) </td> <td class="t-dcl-nopad"> </td> </tr> <tr class="t-dcl t-since-cxx11"> <td> <pre data-language="cpp">explicit underflow_error( const char* what_arg );</pre>
</td> <td> (2) </td> <td> <span class="t-mark-rev t-since-cxx11">(since C++11)</span> </td> </tr>  </table> <p>Constructs the exception object with <code>what_arg</code> as explanatory string that can be accessed through <code><a href="../exception/what" title="cpp/error/exception/what">what()</a></code>.</p>
<p>Because copying <code>std::underflow_error</code> is not permitted to throw exceptions, this message is typically stored internally as a separately-allocated reference-counted string. This is also why there is no constructor taking <code>std::string&amp;&amp;</code>: it would have to copy the content anyway.</p>
<h3 id="Parameters"> Parameters</h3> <table class="t-par-begin"> <tr class="t-par"> <td> what_arg </td> <td> - </td> <td> explanatory string </td>
</tr>
</table> <h3 id="Exceptions"> Exceptions</h3> <p>May throw <code><a href="../../memory/new/bad_alloc" title="cpp/memory/new/bad alloc">std::bad_alloc</a></code>.</p>
</div> <div class="t-inherited"> <h2 id="Inherited_from_std::exception"> Inherited from <a href="../exception" title="cpp/error/exception"> std::exception</a> </h2> <h3 id="Member_functions_2"> Member functions</h3> <table class="t-dsc-begin"> <tr class="t-dsc"> <td> <div><a href="../exception/~exception" title="cpp/error/exception/~exception"> <span class="t-lines"><span>(destructor)</span></span></a></div>
<div><span class="t-lines"><span><span class="t-cmark">[virtual]</span></span></span></div> </td> <td> destroys the exception object <br> <span class="t-mark">(virtual public member function of <code>std::exception</code>)</span> </td>
</tr> <tr class="t-dsc"> <td> <div><a href="../exception/what" title="cpp/error/exception/what"> <span class="t-lines"><span>what</span></span></a></div>
<div><span class="t-lines"><span><span class="t-cmark">[virtual]</span></span></span></div> </td> <td> returns an explanatory string <br> <span class="t-mark">(virtual public member function of <code>std::exception</code>)</span> </td>
</tr> </table> </div>          <div class="_attribution">
  <p class="_attribution-p">
    © cppreference.com<br>Licensed under the Creative Commons Attribution-ShareAlike Unported License v3.0.<br>
    <a href="http://en.cppreference.com/w/cpp/error/underflow_error/underflow_error" class="_attribution-link">http://en.cppreference.com/w/cpp/error/underflow_error/underflow_error</a>
  </p>
</div>

				
			</div>
			<ins class="adsbygoogle"
     style="display:block"
     data-ad-client="ca-pub-2572770204602497"
     data-ad-slot="1992473792"
     data-ad-format="auto"
     data-full-width-responsive="true"></ins>
<script>
     (adsbygoogle = window.adsbygoogle || []).push({});
</script>
		</div>
	</section>

	</div>
	<svg style="display:none">
		<symbol id="icon-dir"><svg viewBox="0 0 20 20"><path d="M15 10c0 .3-.305.515-.305.515l-8.56 5.303c-.625.41-1.135.106-1.135-.67V4.853c0-.777.51-1.078 1.135-.67l8.56 5.305S15 9.702 15 10z"/></svg></symbol>
	  </svg>
</body>
</html>
